src/pyhf/pdf.py

```diff
@@ -422,6 +422,7 @@ def __init__(self, spec, batch_size=None, **config_kwargs):
             if hasattr(parset, 'pdf_type'):  # is constrained
                 self.config.auxdata += parset.auxdata
                 self.config.auxdata_order.append(k)
+        self.config.nauxdata = len(self.config.auxdata)
```

Comment:
why does this get set later instead of at `init` time?

Reply:
because the stuff that `appends` to auxdata is done here see comment above

```
        # this is tricky, must happen before constraint
        # terms try to access auxdata but after
        # combined mods have been created that
        # set the aux data
```

Reply:
i see comment now. yes. this makes sense... not obvious initially.